lib/chef/provider/machine.rb

```diff
@@ -146,6 +146,12 @@ def machine_options(driver)
           result
         end
     }
+    
+    # I'm making the assumption here that you won't be using 2 different drivers
+    # in the same Ruby instance
```

Comment:
I don't think there's another way to do it; and as long as the attribute has the word `aws` in it, I think it's OK for now personally. Use it on a vagrant machine and it just won't do anything.